File: 2345-minimum-number-of-operations-to-convert-time/minimum-number-of-operations-to-convert-time.ts
function convertTime(current: string, correct: string): number {
  function timeToMinutes(time: string): number {
    const [hours, minutes] = time.split(':').map(Number);
    return hours * 60 + minutes;
  }

  let gap = timeToMinutes(correct) - timeToMinutes(current);
  let count = 0;
  const operations = [60, 15, 5, 1];

  for (const op of operations) {
    count += Math.floor(gap / op);
    gap %= op;
  }

  return count;
}

File: 2345-minimum-number-of-operations-to-convert-time/minimum-number-of-operations-to-convert-time.go
import (
    "strconv"
    "strings"
)

func timeToMinutes(time string) int {
    parts := strings.Split(time, ":")
    hours, _ := strconv.Atoi(parts[0])
    minutes, _ := strconv.Atoi(parts[1])
    return hours*60 + minutes
}

func convertTime(current string, correct string) int {
    gap := timeToMinutes(correct) - timeToMinutes(current)
    count := 0
    operations := []int{60, 15, 5, 1}

    for _, op := range operations {
        count += gap / op
        gap %= op
    }

    return count
}
